public/app/features/variables/editor/VariableEditorEditor.tsx

```diff
@@ -110,104 +119,90 @@ export class VariableEditorEditorUnConnected extends PureComponent<Props> {
       return null;
     }
     const loading = variable.state === LoadingState.Loading;
+    const typeOptions = getVariableTypes();
+    const typeValue = typeOptions.find(o => o.value === this.props.variable.type) ?? { label: 'Query', value: 'query' };
+    const typeTooltip = variableAdapters.get(this.props.variable.type).description;
+    const hideOptions = [
```

Comment:
nit: maybe define this one outside of the render function since it looks like it is static and never change.

Reply:
Yepp, those should probably be part of a sub component. I did want to limit the scope of this PR, but maybe I should take it on now instead. Thank you for this suggestion @mckn

Reply:
hmm, maybe a good first step could be just to define them in the top of the file. Then we can move it to their own components later on?